### PR TITLE
UI-test improvements foreman-1.16

### PR DESCRIPTION
**Pull request by m-bucher**
Far away from getting a good pass-rate on foreman-1.16, but these fix some of the problems that occured with foreman-1.16.

### CODE REVIEW

File: robottelo/ui/base.py [hunk 1]
```diff
@@ -455,14 +455,16 @@ def ajax_complete(self, driver):
         angular_active = False
 
         try:
-            jquery_active = driver.execute_script('return jQuery.active') > 0
+            val = driver.execute_script('return jQuery.active')
+            jquery_active = type(val) == int and val > 0
```
Comment (ldjebran) [inline robottelo/ui/base.py]: prefer 
```python
jquery_active =  isinstance(val, int) and val > 0
```

File: robottelo/ui/base.py [hunk 2]
```diff
@@ -455,14 +455,16 @@ def ajax_complete(self, driver):
...
         except WebDriverException:
             pass
 
         try:
-            angular_active = driver.execute_script(
+            val = driver.execute_script(
                 u'return angular.element(document).injector().get("$http")'
-                u'.pendingRequests.length') > 0
+                u'.pendingRequests.length')
+            angular_active = type(val) == int and val > 0
```
Comment (ldjebran) [inline robottelo/ui/base.py]: prefer
```python
angular_active =  isinstance(val, int) and val > 0
```

File: robottelo/ui/navigator.py [hunk 3]
```diff
@@ -12,6 +12,7 @@ def menu_click(self, top_menu_locator, sub_menu_locator,
                    tertiary_menu_locator=None):
         self.perform_action_chain_move(top_menu_locator)
         if not tertiary_menu_locator:
+            self.wait_until_element(sub_menu_locator)
```
Comment (ldjebran) [inline robottelo/ui/navigator.py]: do not see why we have to call explicitly  wait_until_element here as already called in click function

**Comment by ldjebran** [comment]
Please supply tests results for the modified tests, (as we are in master they should be run for 6.4 server)

File: robottelo/ui/navigator.py (hunk 3 above)
Comment (m-bucher) [reply robottelo/ui/navigator.py]: Might have been a problem with my test-setup. I tested with firefox, but as I learned recently there seem to be problems with firefox not waiting for elements.
I can remove this line, if you like.

**Comment by abalakh** [comment]
@omaciel frankly i'm not sure what to do with this PR. It partially fixes legacy robottelo UI automation for foreman-1.16. But we do not support legacy UI automation anymore :)
And, basically, it will be a problem to test these changes - latest selenium is set in requirements, which won't work correctly with old FF42-45, not to mention we don't use FF anymore, so in order to test it out one have to roll back a lot of changes.
I'd vote for closing this PR and recommending to use/contribute to airgun instead :)

**Comment by abalakh** [comment]
@m-bucher yeah, if foreman-1.16 is using horizontal navigation, then sat 6.3.z is your choice. But 6.3.z is using older selenium fully compatible with FF42-47, so most of your changes shouldn't be required there. Pls try it

**Comment by rplevka** [comment]
@m-bucher hey. Are you still interested in this? it's been untouched for some time now

**Comment by m-bucher** [comment]
@rplevka No, since master is vertical-nav anyway, there should be no need to merge a fix for the old-navigation into it.
You may close this PR, thanks.

**Comment by rplevka** [comment]
Thx for confirmation

**Pull request closed without merging**